wgpu/backends/wgpu_native/_api.py

```diff
@@ -2173,16 +2180,15 @@ def read_mapped(self, buffer_offset=None, size=None, *, copy=True):
         src_m = get_memoryview_from_address(src_address, size)
 
         if copy:
-            # Copy the data. The memoryview created above becomes invalid when the buffer
-            # is unmapped, so we don't want to pass that memory to the user.
-            data = memoryview((ctypes.c_uint8 * size)()).cast("B")
-            data[:] = src_m
+            data = memoryview(bytearray(src_m)).cast("B")
```

Comment:
Since data copies should be avoided where possible, but are needed sometimes, I like to be very explicit in comments on what happens and why, even if it may feel obvious sometimes.

The diff for this suggestion looks weird; I only added a comment.
```suggestion
            # Copy the data. The memoryview created above becomes invalid when the buffer is unmapped
            # bytearray() creates a copy, memoryview() creates a view on the byte-array.
            data = memoryview(bytearray(src_m)).cast("B")
```